registry.go

```diff
@@ -16,57 +16,66 @@ package log
 
 import (
 	"sync"
+
+	"github.com/tetratelabs/telemetry"
 )
 
+// Scope is a named logger that can be configured independently
+type Scope interface {
+	telemetry.Logger
+	Name() string
+}
+
 // Register a new logger with the given name.
 // If the logger already exists, the already registered logger is returned.
-func Register(name, description string) *Logger {
-	l := reg.GetLogger(name)
-	if l != nil {
+func Register(name, description string) Scope {
+	if l, ok := reg.GetLogger(name); ok {
 		return l
 	}
 
-	l = newLogger(name, description)
+	l := newLogger(name, description)
 	reg.Register(l)
 
 	return l
 }
 
 // RegisterUnstructured a new unstructured logger with the given name.
 // If the logger already exists, the already registered logger is returned.
-func RegisterUnstructured(name, description string) *Logger {
-	l := reg.GetLogger(name)
-	if l != nil {
+func RegisterUnstructured(name, description string) Scope {
+	if l, ok := reg.GetLogger(name); ok {
 		return l
 	}
 
-	l = newUnstructured(name, description)
+	l := newUnstructured(name, description)
 	reg.Register(l)
 
 	return l
 }
 
 // Loggers returns the list of all registered loggers.
-func Loggers() []*Logger { return reg.Loggers() }
+func Loggers() []Scope { return reg.Loggers() }
 
 // GetLogger gets a registered logger by name.
-func GetLogger(name string) *Logger { return reg.GetLogger(name) }
+func GetLogger(name string) Scope {
```

Comment:
isn't this somewhat a constructor? or we don't call it `NewLogger` because it has side-effects? (registering)

Reply:
It's not. It returns existing loggers, or defaults to the `discard` one.